packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts

```diff
@@ -0,0 +1,300 @@
+        expect(social_share.shareButtonScale).toBe(undefined);
+        //should be defualted to whatever scale is in the render logic
+        social_share.shareButtonScale = 's';
+        expect(social_share.shareButtonScale).toBe('s');
+
+    });
+    //iframeInnerText
+    test('test shareButtonType and setting defualt values', async () => {
+        expect(social_share.iframeInnerText).toBe('');
+       
+        social_share.iframeInnerText = 'This is the test for ifrmae inner text !@';
+        expect(social_share.iframeInnerText).toBe('This is the test for ifrmae inner text !@');
+
+    });
+    //popoverButtonIconScale
+    test('test popoverButtonIconScale and setting defualt values', async () => {
+        expect(social_share.popoverButtonIconScale).toBe('m');
+       
+        social_share.popoverButtonIconScale = 's';
+        expect(social_share.popoverButtonIconScale).toBe('s');
+
+    });
+    test('test displayTipText and setting defualt values', async () => {
+        expect(social_share.displayTipText).toBe(true);
+       
+        social_share.displayTipText = false;
+        expect(social_share.displayTipText).toBe(false);
+
+    });
+    test('test shortenShareUrl and setting defualt values', async () => {
+        expect(social_share.shortenShareUrl).toBe(true);
+       
+        social_share.shortenShareUrl = false;
+        expect(social_share.shortenShareUrl).toBe(false);
+
+    });
+    test('test socialMedia and setting defualt values', async () => {
+        expect(social_share.socialMedia).toBe(true);
+       
+        social_share.socialMedia = false;
+        expect(social_share.socialMedia).toBe(false);
+
+    });
+    test('test shareIconsLayout and setting defualt values', async () => {
+        expect(social_share.shareIconsLayout).toBe('vertical');
+       
+        social_share.shareIconsLayout = 'horizontal';
+        expect(social_share.shareIconsLayout).toBe('horizontal');
+
+    });
+    test('test scale and setting defualt values', async () => {
+        expect(social_share.scale).toBe('m');
+       
+        social_share.scale = 'l';
+        expect(social_share.scale).toBe('l');
+
+    });
+    test('test successMessage and setting defualt values', async () => {
+        expect(social_share.successMessage).toBe('');
+       
+        social_share.successMessage = 'Your awsome map has been copied to clipboard';
+        expect(social_share.successMessage).toBe('Your awsome map has been copied to clipboard');
+
+    });
+    //defaultUrlParams
+    test('test defaultUrlParams and setting defualt values', async () => {
+        expect(social_share.defaultUrlParams).toBe(null);
+        const shareParams =  {
+            center: true,
+            level: true,
+            viewpoint : true,
+            selectedFeature: false,
+            hiddenLayers: true
+        } 
+        social_share.defaultUrlParams = shareParams;
+        
+        expect(social_share.defaultUrlParams).toBe(shareParams);
```

Comment:
if you're testing for values within an object, you'll need to do a more thorough check, try looking through the vitest docs to see how that can be done